packages/freezed/test/integration/multiple_constructors.dart

```diff
@@ -165,3 +165,18 @@ class NestedMapItem with _$NestedMapItem {
   factory NestedMapItem.inner(Map<String, LeafNestedMapItem> children) =
       InnerNestedMapItem;
 }
+
+@freezed
+class ToBeGenerated with _$ToBeGenerated {
+  const factory ToBeGenerated.generated() = CodeGenerated;
```

Comment:
Ideally that `CodeGenerated` should not be a freezed class.

Freezed classes could be special cases, as Freezed could explicitly support those since it should know they are going to be generated